crates/proof-of-sql/utils/commitment-utility/main.rs

```diff
@@ -0,0 +1,135 @@
+    ReadStdin,
+
+    #[snafu(display("Failed to create output file '{:?}'", filename))]
+    CreateOutputFile { filename: PathBuf },
+
+    #[snafu(display("Failed to write to output file '{:?}'", filename))]
+    WriteOutputFile { filename: PathBuf },
+
+    #[snafu(display("Failed to write to stdout"))]
+    WriteStdout,
+
+    #[snafu(display("Failed to deserialize commitment"))]
+    DeserializationError,
+}
+
+type CommitUtilityResult<T, E = CommitUtilityError> = std::result::Result<T, E>;
+
+fn main() -> CommitUtilityResult<()> {
+    let cli = Cli::parse();
+
+    // Read input data
+    let input_data = match &cli.input {
+        Some(input_file) => {
+            let mut file =
+                File::open(input_file).map_err(|_| CommitUtilityError::OpenInputFile {
+                    filename: input_file.clone(),
+                })?;
+            let mut buffer = Vec::new();
+            file.read_to_end(&mut buffer)
+                .map_err(|_| CommitUtilityError::ReadInputFile {
+                    filename: input_file.clone(),
+                })?;
+            buffer
+        }
+        None => {
+            let mut buffer = Vec::new();
+            io::stdin()
+                .read_to_end(&mut buffer)
+                .map_err(|_| CommitUtilityError::ReadStdin)?;
+            buffer
+        }
+    };
+
+    // Deserialize commitment based on the scheme
+    let human_readable = match cli.scheme {
+        CommitmentScheme::DynamicDory => {
+            let commitment: TableCommitment<DynamicDoryCommitment> =
+                postcard::from_bytes(&input_data)
+                    .map_err(|_| CommitUtilityError::DeserializationError)?;
+            format!("{commitment:#?}")
+        }
+        CommitmentScheme::Dory => {
+            let commitment: TableCommitment<DoryCommitment> = postcard::from_bytes(&input_data)
+                .map_err(|_| CommitUtilityError::DeserializationError)?;
+            format!("{commitment:#?}")
+        }
+        CommitmentScheme::Ipa => {
+            let commitment: TableCommitment<RistrettoPoint> = postcard::from_bytes(&input_data)
+                .map_err(|_| CommitUtilityError::DeserializationError)?;
+            format!("{commitment:#?}")
+        }
+    };
+
+    // Write output data
+    match &cli.output {
+        Some(output_file) => {
+            let mut file =
+                File::create(output_file).map_err(|_| CommitUtilityError::CreateOutputFile {
+                    filename: output_file.clone(),
+                })?;
+            file.write_all(human_readable.as_bytes()).map_err(|_| {
+                CommitUtilityError::WriteOutputFile {
+                    filename: output_file.clone(),
+                }
+            })?;
+        }
+        None => {
+            io::stdout()
+                .write_all(human_readable.as_bytes())
+                .map_err(|_| CommitUtilityError::WriteStdout)?;
+        }
+    }
+
+    Ok(())
```

Comment:
is it possible to test this round trip? If not its ok, I know its hard to get all the right files into one place sometimes